### DISCUSSION TITLE
Cannot DrawCircle brighter than background of CreateWindow

### DISCUSSION THREAD

**Discussion by tomasmanu** [bonsai-rx/bonsai#1115] [Q&A] [answered]
Hi all!
 I am trying to draw looming stimuli brighter than the background (ON stimuli), but the `ColorA` scale of `DrawCircle` does not work for this situation, as I assume that this scale is calculated as `1 - (RGB stimulus / RGB background)
`

![image](https://user-images.githubusercontent.com/69548554/200031983-115a1d78-698c-4160-83bb-45797e4d376f.png)

How can I solve this problem? 

I would also want to ask if there is a way of modifying the color of `CreateWindow` in a continuous manner rather than selecting the color from a set of available colors.

![image](https://user-images.githubusercontent.com/69548554/200032165-d5b7f7b4-2eaf-477e-895e-8ee6d4836272.png)

Thanks!

**Comment by bruno-f-cruz** [comment, marked as answer]
"A" is actually "Alpha", it controls the transparency of the object. I am not sure what the default blending method is right now but I believe it is just weighting the object and everything below. In your case, I believe it will be something like A * RGB_top + (1-A) * RGB_Behind.
If you want to dynamically change the background, I would just throw a Quad behind the circle, which will allow you to easily change the color whenever you want.

**Reply by tomasmanu** [reply]
Thank you Bruno! The alternative of the Quad behind the circle worked perfectly. And respect to the brighter stimulus, reading your answer I realised that I did not change the RGB values of `DrawCircle`, what is the key to generate the kind of stimuli I needed!

**Comment by aaronjnorris** [comment]
any chance you would be willing to share the script?

**Reply by tomasmanu** [reply]
Of course! I leave it in the zip attached.
[color_DrawCircle.zip](https://github.com/bonsai-rx/bonsai/files/9966254/color_DrawCircle.zip)
P.D: the looming used here is not a 'real looming', I used that one just to make it simple, because the nature of the stimulus was not important for the question. But if you wanted the workflow because of the stimulus, here is a post in the Bonsai Google group where I left a workflow with a more appropriate looming dynamic: https://groups.google.com/g/bonsai-users/c/yHiums83FAI/m/26aCzyRFAAAJ
